docs/porting.rst

```diff
@@ -42,6 +42,43 @@ to the port's directory (in the top level until the ``ports`` directory is
 present). This includes the Makefile and any C library resources. Make sure
 these resources are compatible with the MIT License of the rest of the code!
 
+Circuitpython has a number of modules enabled by default in
+``py/circuitpy_mpconfig.mk``. Most of these modules will need to be disabled in
+``mpconfigboard.mk`` during the early stages of a port in order for it to
+compile. As the port progresses in module support, this list can be pruned down
+as a natural "TODO" list. An example minimal build list is shown below:
+
+.. code-block:: makefile
+
+    # Items requiring ports/<port>/common-hal implementation:
+    CIRCUITPY_MICROCONTROLLER = 0 # Typically the first module to create
+    CIRCUITPY_DIGITALIO = 0       # Typically the second module to create
+    CIRCUITPY_ANALOGIO = 0
+    CIRCUITPY_BUSIO = 0
+    CIRCUITPY_NEOPIXEL_WRITE = 0
+    CIRCUITPY_PULSEIO = 0
+    CIRCUITPY_OS = 0
+    CIRCUITPY_NVM = 0
+    CIRCUITPY_AUDIOBUSIO = 0
+    CIRCUITPY_AUDIOIO = 0
+    CIRCUITPY_ROTARYIO = 0
+    CIRCUITPY_RTC = 0
+    CIRCUITPY_FREQUENCYIO = 0
+    CIRCUITPY_I2CSLAVE = 0
+    CIRCUITPY_DISPLAYIO = 0       # Requires SPI, PulseIO (stub ok)
+
+    # Modules with no common-hal implementation, but depend on something else
+    CIRCUITPY_BITBANGIO = 0       # Requires DigitalIO
+    CIRCUITPY_GAMEPAD = 0         # Requires DigitalIO
+    CIRCUITPY_PIXELBUF = 0        # Does nothing without a neopixel or dotstar
+    CIRCUITPY_RANDOM = 0          # Requires OS
+    CIRCUITPY_STORAGE = 0         # Requires OS, filesystem
+    CIRCUITPY_TOUCHIO = 0         # Requires Microcontroller
+    CIRCUITPY_USB_HID = 0         # Requires USB
+    CIRCUITPY_USB_MIDI = 0        # Requires USB
+    CIRCUITPY_REQUIRE_I2C_PULLUPS = 0 # Does nothing without I2C
+    CIRCUITPY_ULAB = 0            # No requirements, but takes extra flash
```

Comment:
These are implemented in `shared-module`, and are port-independent, so as soon as their port-dependent dependencies are implemented, they can be turned on. So could you change the comment to reflect that? For `CIRCUITPY_PIXELBUF`, it doesn't have dependencies, but it's not useful until either `neopixel_write` (NeoPixels) or `SPI` (DotStars) are available.

Reply:
Good idea.